dvc/remote/azure.py

```diff
@@ -15,56 +14,17 @@
 
 
 class AzureRemoteTree(BaseRemoteTree):
```

Comment:
Would it make sense to make these just `Azure` or `AzureTree` and move them to `dvc/tree/azure.py` instead of continually thinking about them as remotes?

Reply:
I think it's worth considering if we are going to start using these tree's for non-remote/cache related use cases (like workspaces).

Reply:
We do that already for external outputs scenario :) Workspaces are a just a potential better abstraction for that.

But ok, let's merge and then do that later.